lib/rules/no-array-constructor.js

```diff
@@ -49,6 +51,70 @@ module.exports = {
 	create(context) {
 		const sourceCode = context.sourceCode;
 
+		/**
+		 * Checks whether comments exist between the array constructor and its elements.
+		 * This function examines different parts of an array constructor node to determine if any comments are present in the following locations:
+		 * - Between the `new` keyword and the `Array` identifier.
+		 * - Between the opening parenthesis `(` and the `Array` identifier.
+		 * - Between the `Array` identifier and the closing parenthesis `)`.
+		 * - Between the closing parenthesis `)` and the opening parenthesis `(` of the arguments.
+		 * @param {ASTNode} node The array constructor node to check. This can be either a
+		 * `CallExpression` (for `Array()`) or a `NewExpression` (for `new Array()`).
+		 * @returns {boolean} `true` if comments exist in any of the checked locations, `false` otherwise.
+		 */
+		function hasCommentsInArrayConstructor(node) {
+			const calleeToken = node.callee;
+
+			if (node.type === "NewExpression") {
+				const newToken = sourceCode.getFirstToken(node);
+				if (sourceCode.commentsExistBetween(newToken, calleeToken)) {
+					return true;
+				}
+			}
+
+			let lastRelevantToken = calleeToken;
+
+			const tokenBefore = sourceCode.getTokenBefore(calleeToken);
+			if (tokenBefore && isOpeningParenToken(tokenBefore)) {
+				if (sourceCode.commentsExistBetween(tokenBefore, calleeToken)) {
+					return true;
+				}
+
+				const tokenAfter = sourceCode.getTokenAfter(calleeToken);
+				if (tokenAfter && isClosingParenToken(tokenAfter)) {
+					if (
+						sourceCode.commentsExistBetween(calleeToken, tokenAfter)
+					) {
+						return true;
+					}
+					lastRelevantToken = tokenAfter;
+				}
+			}
+
+			const lastToken = sourceCode.getLastToken(node);
+			let currentToken = lastRelevantToken;
+
+			do {
+				currentToken = sourceCode.getTokenAfter(currentToken);
+				if (!currentToken || currentToken === lastToken) {
+					break;
+				}
+			} while (!isOpeningParenToken(currentToken));
```

Comment:
This could end up searching through tokens after the call/new expression, and provide a suggestion instead of autofix.

For example:

```js
const arr = new Array;

// this comment is found, although it's after the expression

foo( // <-- this opening paren is found

);

```